ui/src/app/applications/components/applications-sync-panel/applications-sync-panel.tsx

```diff
@@ -33,26 +40,30 @@ export const ApplicationsSyncPanel = ({show, apps, hide}: {show: boolean; apps:
                                 ctx.notifications.show({content: `No apps selected`, type: NotificationType.Error});
                                 return;
                             }
-                            ctx.notifications.show({
-                                content: `Syncing ${selectedApps.length} app(s)`,
-                                type: NotificationType.Success
-                            });
                             const syncStrategy = (params.applyOnly ? {apply: {force: params.force}} : {hook: {force: params.force}}) as models.SyncStrategy;
+                            setProgress({percentage: 0, title: 'Starting...'});
+                            let i = 0;
                             for (const app of selectedApps) {
                                 await services.applications.sync(app.metadata.name, app.spec.source.targetRevision, params.prune, params.dryRun, syncStrategy, null).catch(e => {
                                     ctx.notifications.show({
                                         content: <ErrorNotification title={`Unable to sync ${app.metadata.name}`} e={e} />,
                                         type: NotificationType.Error
                                     });
                                 });
+                                i++;
+                                setProgress({
+                                    percentage: i / selectedApps.length,
+                                    title: `${i} of ${selectedApps.length} apps synced`
+                                });
                             }
-                            hide();
+                            setProgress({percentage: 100, title: 'Complete'});
                         }}
                         getApi={setForm}>
                         {formApi => (
                             <React.Fragment>
                                 <div className='argo-form-row'>
                                     <h4>Sync app(s)</h4>
+                                    {progress !== null && <ProgressPopup onClose={() => setProgress(null)} percentage={progress.percentage} title={progress.title} />}
```

Comment:
What happens if user close progess popup in a middle of syncing operation? Once the next app sync starts progress popup will be shown again?

Reply:
You can't. The pop-up is modal. Nor can you close the pop-up. You can close the window, which aborts the sync.

Reply:
Question: do we need a better way to abort?

Reply:
I think it is good idea. Does not have to be part of the same PR though.